### PR TITLE
Check dates against `DateTimeInterface` instead of `DateTime`

### PR DESCRIPTION
**Pull request by jeromegamez**
Greeting! 👋 

I noticed that `where()` queries in which I used `DateTimeImmutable` objects returned empty results.

This change would allow the usage of `DateTimeImmutable` or `CarbonImmutable` in addition to `DateTime`/`Carbon` 🤞 

:octocat:

### CODE REVIEW

File: tests/QueryBuilderTest.php [hunk 1]
```diff
@@ -574,6 +574,33 @@ public function testDates()
...
 
+    public function testImmutableDates()
+    {
+        DB::collection('users')->insert([
+            ['name' => 'John Doe', 'birthday' => new UTCDateTime(Date::parse("1980-01-01 00:00:00")->format('Uv'))],
+            ['name' => 'Robert Roe', 'birthday' => new UTCDateTime(Date::parse("1982-01-01 00:00:00")->format('Uv'))],
+        ]);
+
+        $user = DB::collection('users')->where('birthday', '=', new DateTimeImmutable("1980-01-01 00:00:00"))->first();
+        $this->assertNotNull($user);
```
Comment (Smolevich) [inline tests/QueryBuilderTest.php:585]: Add `assertCount` for check count elements

**Review by Smolevich — COMMENTED** [review]
Needs work

File: tests/QueryBuilderTest.php (hunk 1 above)
Comment (jeromegamez) [reply tests/QueryBuilderTest.php]: Changed as requested ✅

**Review by divine — APPROVED** [review]

**Comment by jeromegamez** [comment]
Shall I address the StyleCI issues in a separate commit?

**Comment by divine** [comment]
> Shall I address the StyleCI issues in a separate commit?

Hello,

No, we'll take care of it later.

Thanks!

**Review by Smolevich — APPROVED** [review]

**Comment by divine** [comment]
Thank you for your contribution @jeromegamez !

**Pull request merged**